pkg/model/deployment_chain.proto

```diff
@@ -0,0 +1,48 @@
+// Copyright 2021 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package pipe.model;
+option go_package = "github.com/pipe-cd/pipe/pkg/model";
+
+import "validate/validate.proto";
+import "pkg/model/application.proto";
+import "pkg/model/common.proto";
+
+message DeploymentChain {
+    // The generated unique identifier.
+    string id = 1 [(validate.rules).string.min_len = 1];
+    // The ID of the project this environment belongs to.
+    string project_id = 2 [(validate.rules).string.min_len = 1];
+    // Deployment node contains all configuration required to perform deployment(s)
+    repeated DeploymentNode nodes = 3;
+    // Unix time when all the applications in this chain are deployed.
+    int64 completed_at = 100 [(validate.rules).int64.gte = 0];
+    // Unix time when the deployment chain is created.
+    int64 created_at = 101 [(validate.rules).int64.gt = 0];
+    // Unix time of the last time when the deployment chain is updated.
+    int64 updated_at = 102 [(validate.rules).int64.gt = 0];
+}
+
+message DeploymentNode {
+    // List of applications which should be deployed at the same time in chain.
+    repeated Application applications = 1;
```

Comment:
I think we have to think about this field.
Should it be a list of applications or a list of triggered deployments or a list of application IDs?
Could you share your thoughts about this?

Reply:
Btw, in the case of storing the application list, how can we have the status (info) of actually triggered deployments to show in the web console at the "Chains" tab?

Reply:
Ah, I get your point 👍  Since the name of this model is `DeploymentChain` it should be a chain of deployment, that what I was thought at first, but it looks like at the time we create this DeploymentChain model object, we don't trigger for deployment yet so that we don't have deployment object to use, my plan is to use ApplicationDeploymentReference. But I was wrong, we may need to add ability to piped of the first application in chain to enable it to trigger deployments for all application in chain and use it here 👍

Reply:
Thanks for your valuable comment 🙏 Addressed by https://github.com/pipe-cd/pipe/pull/2790/commits/668b6644b3cf347b4713e36a26d8d241bd308a74

Reply:
> we may need to add the ability to piped of the first application in the chain to enable it to trigger deployments for all applications in the chain and use it here

Actually, I still don't see a clear path for triggering all deployments from the first piped since some applications could be handled by other piped.
But it seems we have to store both application ref and deployment ref: all application ref are added when the deployment chain model is initialized, deployment refs are added when its piped triggers them.

I think it is hard to find out a perfect one at this time. So I am ok to merge this PR now and we can update the model if needed later.